mcs/class/referencesource/mscorlib/system/threading/threadpool.cs

```diff
@@ -1507,7 +1507,10 @@ bool               compressStack
                 throw new NotSupportedException ("Timeout is too big. Maximum is Int32.MaxValue");
 
             RegisteredWaitHandle waiter = new RegisteredWaitHandle (waitObject, callBack, state, new TimeSpan (0, 0, 0, 0, (int) millisecondsTimeOutInterval), executeOnlyOnce);
-            QueueUserWorkItem (new WaitCallback (waiter.Wait), null);
+			if (compressStack)
```

Comment:
This should be between `#if MONO`

Reply:
It already is, see line 1471.

Fixing the indentation to match the surrounding code would be good though :)